CognitiveVRTest/Plugins/CognitiveVR/Source/CognitiveVR/Private/network/network.cpp

```diff
@@ -169,16 +259,43 @@ void Network::OnLocalCacheCallReceivedAsync(FHttpRequestPtr Request, FHttpRespon
 				localCacheRequest->SetHeader("Content-Type", TEXT("application/json"));
 				localCacheRequest->SetHeader("Authorization", AuthValue);
 
-				localCacheRequest->OnProcessRequestComplete().BindRaw(this, &Network::OnLocalCacheCallReceivedAsync);
-				localCacheRequest->ProcessRequest();
+				world->GetTimerManager().ClearTimer(TimerHandleShortDelay);
+				world->GetTimerManager().SetTimer(TimerHandleShortDelay, FTimerDelegate::CreateRaw(this, &Network::RequestLocalCache), LocalCacheLoopDelay, false);
+
 			}
 			else
 			{
 				localCacheRequest = NULL;
+				world->GetTimerManager().ClearTimer(TimerHandleShortDelay);
 			}
 		}
 		else
 		{
+			if (responseCode < 500 && responseCode != 0) 
+			{ 
+				cog->localCache->PopContent();
+				world->GetTimerManager().ClearTimer(TimerHandleShortDelay);
+				localCacheRequest = NULL;
+				return; 
+			}
+			
+
+			//if local cache upload request gets 500 response start variable delay timer
+			if (!IsServerUnreachable)
+			{
+				IsServerUnreachable = true;
+				VariableDelayMultiplier++;
```

Comment:
Very very small chance of an int32 overflowing but maybe put a check here just to be safe? Good practice anyway. 

What I'm getting at is what happens to a `++` on an int32 as it reaches the highest possible value?

On second thought, maybe just clamp the delay at a smaller multiplier? Like `10` or something? No need to ever delay longer than 10 minutes? (Or maybe 20 minutes?) My point is that it doesn't need to be 1,000,000 minutes.